controllers/construct/mongodbstatefulset.go

```diff
@@ -57,6 +57,7 @@ type MongoDBStatefulSetOwner interface {
 	AutomationConfigSecretName() string
 	GetUpdateStrategyType() appsv1.StatefulSetUpdateStrategyType
 	Persistent() bool
+	HasSeparateDataAndLogsVolumes() bool
```

Comment:
[nit] can we add some docstrings to the methods in this interface since we're modifying it?